cmake/Modules/HandleCUDAComputeCapability.cmake

```diff
@@ -8,24 +8,44 @@
 # those without the CUDA_ prefix
 
 if (NOT CUDA_TARGET_COMPUTE_CAPABILITY)
-	if("$ENV{CUDA_SM}" STREQUAL "")
-		set(ENV{CUDA_INCLUDE_DIRS} "${CUDA_INCLUDE_DIRS}")
-		set(ENV{CUDA_CUDART_LIBRARY} "${CUDA_CUDART_LIBRARY}")
-		set(ENV{CMAKE_CXX_COMPILER} "${CMAKE_CXX_COMPILER}")
-		execute_process(COMMAND bash -c "${CMAKE_CURRENT_SOURCE_DIR}/scripts/get_cuda_sm.sh"  OUTPUT_VARIABLE CUDA_TARGET_COMPUTE_CAPABILITY_) 
+	if(WIN32)
+		find_package(CUDA)
+		if (CUDA_FOUND)
+			set(CUDAFILE ${CMAKE_CURRENT_SOURCE_DIR}/examples/check_cuda.cu)
```

Comment:
It seems you've developed a facility for detecting the CUDA SM on Windows. Please:

1. Make that a separate PR
2. Put it in a CMake module
3. If you can spare the time - See how much of the Unix-oriented shell script can be discarded in favor of what you do in CMake, so that the code paths will not look so different.

Reply:
regarding 3. - I do not see a reason why we couldn't use the CMake version on Unix as well. - I simply didn't delete it and made the Windows stuff conditional since I can't test if it has exactly the same behaviour.